### ISSUE TITLE
Improve Flag activities to include the categorisation of the report

### ISSUE THREAD

**Issue by ThisIsMissEm** [swicg/activitypub-trust-and-safety#2]
Currently `Flag` activities do not contain the categorisation of a report, which means that reports that are federated from other servers just show up as "other"

A work around was proposed in Mastodon by abusing `summary` and concatenating `content` for distributing this information, but we may want something more structured here.

**Comment by ThisIsMissEm** [comment]
Example issues in Mastodon relating to this:
- https://github.com/mastodon/mastodon/issues/31604
- https://github.com/mastodon/mastodon/issues/31092

**Comment by AnSavvides** [comment]
This would be great - also having a set of pre-defined labels (probably with ways to add custom ones too if need be) like we discussed at FediForum in the past would be really helpful, especially for things like reporting CSAM via Flag activities and knowing there is a very clear and consistent way to mark that a Flag is pertaining to CSAM.

**Comment by ThisIsMissEm** [comment]
I'm still writing the content labelling section in the FIRES documentation, but I'm thinking of leaning on AS2 structuring for the JSON form: https://fires.fedimod.org/concepts/terms/label-provider.html

**Comment by ThisIsMissEm** [comment]
I think there'll end up being a few different vocabularies, and per-instance rules could also be their own vocabularies too.

But I'd expect probably a limited number of top level vocabularies, otherwise the reporting UI becomes horrific — see Twitter's report UI in Germany, it's impossible to navigate

**Comment by AnSavvides** [comment]
> But I'd expect probably a limited number of top level vocabularies

Yeah I definitely agree on trying to keep these very limited / small; trying to account for _everything_, where there is nuance based on location, language, preferences and other such things will be impossible 😅

**Comment by ThisIsMissEm** [comment]
I mean, someone could create a vocabulary that is specific to them, that's _fine_ but in order for software to understand it in a non-generic manner, it'd probably need to draw relationships between itself and more widely understood vocabularies.

As for language, given we can localize contents in AS2, I think this would be possible (e.g., the one vocabulary but available in multiple languages?)
